File: Week2/KimSungHa/B1259.py
# 문제: 1259번
# 설명: 문자를 입력받아 그 단어가 팰린드롬이면 yes, 아니면 no를 출력하시오
# 테스트 케이스로 볼 때, 0을 입력하면 반복 중단으로 보여 while로 구상

# word = input() 

# if word == '0': 
#     pass
# elif word == word[::-1]: 
#     print('yes')
# else:
#     print('no')

while True: # 이후 0을 받지 않았다면 반복
    word = input()  # 입력받은 단어 및 정수를 문자열로 할당

    if word == '0': # 0을 입력하면 중단이므로 첫 번째로 따져주고,
        break
    elif word == word[::-1]:    # 팰린드롬 여부 판단하여 조건 별 출력
        print('yes')
    else:
        print('no')


# 동일하게 두 문단을 쓰지 않고도 0이면 중단하는 루프를 만들 수 있을까? 해결!

File: Week2/송영재/B1259.java
package problem;

import java.io.IOException;
import java.util.Scanner;

public class B1259 {
    public static void main(String[] args) throws IOException{
        Scanner sc = new Scanner(System.in);
		
        // 입력이 0 누를때까지 이니까 while문을 반복합니다.
		while (true) {
			int input = sc.nextInt(); // 처음에 숫자로 받습니다.
			String num = Integer.toString(input); // 이게 펠롬드린수를 확인하려면 숫자를 문자열로 바꿉니다.
			int N = num.length(); // 이건 String의 길이를 변수에 저장해놓습니다.
			char[] c = num.toCharArray(); // 받은 정수를 문자열로 바꾼 것을 하나의 문자배열에 넣습니다.
			boolean isOk = true; // 이건 펠롬드린 수가 맞나? 확인하는 flag역할입니다.
			
			for (int i = 0; i < N/2; i++) { // 맨끝 처음 그리고 2번째 맨마지막-1번째 확인을 하기 위해 결국 전체길이의 반만 돌면 됩니다.
				if (c[i] != c[N-1-i]) { // 이걸로 처음과 끝을 확인합니다.
					isOk = false; // 만약 처음과 끝이 다르면 더 확인할 필요도 없으니까 isOk를 false로 만들어줍니다.
					break;    // 펠롬드린 수가 아닌 친구는 isOk를 false로 반환합니다
				}
			}
			if (isOk && input != 0) { // 결국 출력을 하려면 isOk가 true여야 펠롬드린 수이므로 true와 0이 아니면 이를 실행합니다.
				System.out.println("yes");
			} else if (isOk == false && input != 0){ // 위와 비슷한 매커니즘입니다.
				System.out.println("no");
			}
			
			
			
			if (input == 0) {
				break;
			}
		}
    }
}
